admin/javascript/LeftAndMain.BatchActions.js

```diff
@@ -319,8 +420,10 @@
 					btn.attr('disabled', 'disabled').button('refresh');
 				} else {
 					btn.removeAttr('disabled').button('refresh');
-					// form.submit();
-				} 
+				}
+				
+				// Refresh selected / enabled nodes
+				$('#Form_BatchActionsForm').refreshSelected();
```

Comment:
This code was removed years ago; Re-added this in to ensure tree items are properly filtered.